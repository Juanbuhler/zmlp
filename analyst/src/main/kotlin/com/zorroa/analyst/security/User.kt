package com.zorroa.analyst.security

data class JobCredentials (val accessKey: String, val secretKey: String)



