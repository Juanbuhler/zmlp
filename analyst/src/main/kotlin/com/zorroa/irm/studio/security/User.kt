package com.zorroa.irm.studio.security

data class JobCredentials (val accessKey: String, val secretKey: String)



