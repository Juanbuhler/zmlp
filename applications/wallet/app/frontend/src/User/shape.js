import PropTypes from 'prop-types'

const userShape = {
  id: PropTypes.number.isRequired,
  username: PropTypes.string.isRequired,
  email: PropTypes.string.isRequired,
  first_name: PropTypes.string.isRequired,
  last_name: PropTypes.string.isRequired,
}

export default userShape
