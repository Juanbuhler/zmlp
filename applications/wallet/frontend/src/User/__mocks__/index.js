import { createContext, createElement } from 'react'

import user from './user'

export const noop = () => {}

export const UserContext = createContext({
  user,
  googleAuth: {},
  setGoogleAuth: noop,
})

const User = ({ children, ...rest }) => createElement('User', rest, children)

export default User
