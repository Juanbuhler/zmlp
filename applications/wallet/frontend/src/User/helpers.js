export const noop = () => {}

export const meFetcher =
  typeof window === 'undefined'
    ? noop
    : async (url) => {
        try {
          const response = await fetch(url)
          if (response.status >= 400) throw response
          return response.json()
        } catch (error) {
          return {}
        }
      }
