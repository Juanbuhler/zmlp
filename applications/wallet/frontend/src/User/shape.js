import PropTypes from 'prop-types'

const userShape = {
  id: PropTypes.number.isRequired,
  username: PropTypes.string.isRequired,
  email: PropTypes.string.isRequired,
  firstName: PropTypes.string.isRequired,
  lastName: PropTypes.string.isRequired,
  organizations: PropTypes.arrayOf(PropTypes.string).isRequired,
}

export default userShape
