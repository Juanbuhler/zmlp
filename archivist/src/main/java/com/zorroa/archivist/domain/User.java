package com.zorroa.archivist.domain;

import com.google.common.base.MoreObjects;
import com.google.common.hash.HashCode;


import java.io.Serializable;

public class User implements Loggable<Integer>, Serializable {

    private int id;
    private String username;
    private String firstName;
    private String lastName;
    private String email;
    private Boolean enabled;
    private UserSettings settings;

    public int getId() {
        return id;
    }

    public User setId(int id) {
        this.id = id;
        return this;
    }

    public String getUsername() {
        return username;
    }

    public User setUsername(String username) {
        this.username = username;
        return this;
    }

    public String getFirstName() {
        return firstName;
    }

    public User setFirstName(String firstName) {
        this.firstName = firstName;
        return this;
    }

    public String getLastName() {
        return lastName;
    }

    public User setLastName(String lastName) {
        this.lastName = lastName;
        return this;
    }

    public String getEmail() {
        return email;
    }

    public User setEmail(String email) {
        this.email = email;
        return this;
    }

    public Boolean getEnabled() {
        return enabled;
    }

    public User setEnabled(Boolean enabled) {
        this.enabled = enabled;
        return this;
    }

    public UserSettings getSettings() {
        return settings;
    }

    public User setSettings(UserSettings settings) {
        this.settings = settings;
        return this;
    }

    @Override
    public boolean equals(Object obj) {
        if (obj == null) { return false; }
        if (!(obj instanceof User)) {
            return false;
        }
        User other = (User) obj;
        return other.getId() == getId();
    }

    @Override
    public int hashCode() {
        return HashCode.fromInt(getId()).hashCode();
    }

    @Override
    public String toString() {
        return MoreObjects.toStringHelper(getClass())
                .add("id", getId())
                .add("username", getUsername())
                .toString();
    }

    @Override
    public Integer getTargetId() {
        return id;
    }
}
