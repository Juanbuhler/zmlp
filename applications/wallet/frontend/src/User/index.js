import { createContext, useState, useEffect } from 'react'
import PropTypes from 'prop-types'
import * as Sentry from '@sentry/browser'

import { getUser, setUser } from './helpers'

export const noop = () => {}

export const UserContext = createContext({
  user: {},
  setUser: noop,
  googleAuth: {},
  setGoogleAuth: noop,
})

const User = ({ initialUser, children }) => {
  const [user, setStateUser] = useState(initialUser)
  const [hasLocalStorageLoaded, setHasLocalStorageLoaded] = useState(false)
  const [googleAuth, setGoogleAuth] = useState({ signIn: noop, signOut: noop })

  useEffect(() => {
    if (initialUser.id || hasLocalStorageLoaded) return

    const storedUser = getUser()

    setStateUser(storedUser)

    setHasLocalStorageLoaded(true)

    /* istanbul ignore next */
    Sentry.configureScope(scope => {
      scope.setUser(storedUser)
    })
  }, [initialUser, hasLocalStorageLoaded, user, setStateUser])

  if (!initialUser.id && !hasLocalStorageLoaded) return null

  return (
    <UserContext.Provider
      value={{
        user,
        setUser: setUser({ setStateUser, user }),
        googleAuth,
        setGoogleAuth,
      }}>
      {children}
    </UserContext.Provider>
  )
}

User.propTypes = {
  initialUser: PropTypes.shape({
    id: PropTypes.number,
    email: PropTypes.string,
  }).isRequired,
  children: PropTypes.node.isRequired,
}

export default User
