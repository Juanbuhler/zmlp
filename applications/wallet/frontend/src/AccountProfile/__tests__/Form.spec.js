import TestRenderer, { act } from 'react-test-renderer'

import User from '../../User'

import AccountProfileForm from '../Form'

jest.mock('../helpers')

const noop = () => () => {}

describe('<AccountProfileForm />', () => {
  it('should render properly when a user edits their name', () => {
    const component = TestRenderer.create(
      <User
        initialUser={{
          id: 1,
          username: 'jane.doe',
          email: 'jane.doe@zorroa.com',
        }}>
        <AccountProfileForm />
      </User>,
    )

    expect(component.toJSON()).toMatchSnapshot()

    // Click edit
    act(() => {
      component.root
        .findByProps({ children: 'Edit' })
        .props.onClick({ preventDefault: noop })
    })

    expect(component.toJSON()).toMatchSnapshot()

    act(() => {
      component.root
        .findByProps({ id: 'firstName' })
        .props.onChange({ target: { value: 'Jane' } })
    })

    act(() => {
      component.root
        .findByProps({ id: 'lastName' })
        .props.onChange({ target: { value: 'Doe' } })
    })

    // Submit the form
    act(() => {
      component.root
        .findByProps({ children: 'Save' })
        .props.onClick({ preventDefault: noop })
    })

    expect(component.toJSON()).toMatchSnapshot()
  })

  it('should render properly when a user cancels', () => {
    const component = TestRenderer.create(
      <User
        initialUser={{
          id: 1,
          username: 'jane.doe',
          email: 'jane.doe@zorroa.com',
        }}>
        <AccountProfileForm />
      </User>,
    )

    // Click edit
    act(() => {
      component.root
        .findByProps({ children: 'Edit' })
        .props.onClick({ preventDefault: noop })
    })

    // Click cancel
    act(() => {
      component.root
        .findByProps({ children: 'Cancel' })
        .props.onClick({ preventDefault: noop })
    })

    expect(component.toJSON()).toMatchSnapshot()
  })
})
