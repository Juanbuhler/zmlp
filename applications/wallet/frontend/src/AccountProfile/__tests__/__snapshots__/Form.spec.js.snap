// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AccountProfileForm /> should render properly when a user cancels 1`] = `
Array [
  .emotion-0 {
  font-size: 17px;
  line-height: 20px;
  font-weight: 500;
  padding-top: 16px;
}

<h3
    className="emotion-0"
  >
    Email: jane.doe@zorroa.com
  </h3>,
  .emotion-0 {
  padding-top: 24px;
  font-size: 17px;
  line-height: 20px;
  font-weight: 500;
}

<div
    className="emotion-0"
  >
    Name
  </div>,
  .emotion-0 {
  color: #808080;
  font-size: 15px;
  line-height: 19px;
  font-weight: 400;
  padding-bottom: 24px;
}

<div
    className="emotion-0"
  >
     
  </div>,
  .emotion-0 {
  -webkit-flex: 0;
  -ms-flex: 0;
  flex: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-radius: 2px;
  margin: 0;
  padding: 12px 32px;
  white-space: nowrap;
  font-size: 15px;
  line-height: 19px;
  font-weight: 500;
  border: none;
  cursor: pointer;
  color: #ffffff;
  background-color: transparent;
}

.emotion-0:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-0[aria-disabled=true] {
  color: #2e2e2e;
}

.emotion-0,
.emotion-0:hover,
.emotion-0:visited {
  background-color: #73b61c;
}

.emotion-0:hover {
  background-color: #96c955;
}

.emotion-0[aria-disabled=true] {
  background-color: #808080;
}

<button
    className="emotion-0"
    onClick={[Function]}
    type="button"
  >
    Edit
  </button>,
]
`;

exports[`<AccountProfileForm /> should render properly when a user edits their name 1`] = `
Array [
  .emotion-0 {
  font-size: 17px;
  line-height: 20px;
  font-weight: 500;
  padding-top: 16px;
}

<h3
    className="emotion-0"
  >
    Email: jane.doe@zorroa.com
  </h3>,
  .emotion-0 {
  padding-top: 24px;
  font-size: 17px;
  line-height: 20px;
  font-weight: 500;
}

<div
    className="emotion-0"
  >
    Name
  </div>,
  .emotion-0 {
  color: #808080;
  font-size: 15px;
  line-height: 19px;
  font-weight: 400;
  padding-bottom: 24px;
}

<div
    className="emotion-0"
  >
     
  </div>,
  .emotion-0 {
  -webkit-flex: 0;
  -ms-flex: 0;
  flex: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-radius: 2px;
  margin: 0;
  padding: 12px 32px;
  white-space: nowrap;
  font-size: 15px;
  line-height: 19px;
  font-weight: 500;
  border: none;
  cursor: pointer;
  color: #ffffff;
  background-color: transparent;
}

.emotion-0:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-0[aria-disabled=true] {
  color: #2e2e2e;
}

.emotion-0,
.emotion-0:hover,
.emotion-0:visited {
  background-color: #73b61c;
}

.emotion-0:hover {
  background-color: #96c955;
}

.emotion-0[aria-disabled=true] {
  background-color: #808080;
}

<button
    className="emotion-0"
    onClick={[Function]}
    type="button"
  >
    Edit
  </button>,
]
`;

exports[`<AccountProfileForm /> should render properly when a user edits their name 2`] = `
Array [
  .emotion-0 {
  font-size: 17px;
  line-height: 20px;
  font-weight: 500;
  padding-top: 16px;
}

<h3
    className="emotion-0"
  >
    Email: jane.doe@zorroa.com
  </h3>,
  .emotion-13 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 470px;
  padding: 16px;
  padding-left: 0;
  padding-right: 0;
}

.emotion-3 {
  padding-top: 8px;
}

.emotion-0 {
  display: block;
  padding-bottom: 8px;
  color: #b3b3b3;
}

.emotion-2 {
  position: relative;
}

.emotion-1 {
  outline-offset: 0;
  padding: 10px 12px;
  border-radius: 2px;
  box-shadow: inset 0 1px 3px 0 transparent;
  width: 100%;
  color: #ffffff;
  background-color: #3b3b3b;
  border: 1px solid transparent;
  border-width: 2px;
}

.emotion-1:focus {
  border: 2px solid #73b61c;
  outline: #73b61c;
}

.emotion-1:focus {
  color: #000000;
  background-color: #ffffff;
}

.emotion-12 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding-top: 16px;
  padding-bottom: 16px;
}

.emotion-12 button,
.emotion-12 a {
  margin-right: 16px;
}

.emotion-10 {
  -webkit-flex: 0;
  -ms-flex: 0;
  flex: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-radius: 2px;
  margin: 0;
  padding: 12px 32px;
  white-space: nowrap;
  font-size: 15px;
  line-height: 19px;
  font-weight: 500;
  border: none;
  cursor: pointer;
  color: #ffffff;
  background-color: transparent;
}

.emotion-10:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-10[aria-disabled=true] {
  color: #2e2e2e;
}

.emotion-10,
.emotion-10:hover,
.emotion-10:visited {
  background-color: #808080;
}

.emotion-10:hover {
  background-color: #b3b3b3;
}

.emotion-10[aria-disabled=true] {
  background-color: #808080;
}

.emotion-11 {
  -webkit-flex: 0;
  -ms-flex: 0;
  flex: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-radius: 2px;
  margin: 0;
  padding: 12px 32px;
  white-space: nowrap;
  font-size: 15px;
  line-height: 19px;
  font-weight: 500;
  border: none;
  cursor: not-allowed;
  color: #ffffff;
  background-color: transparent;
}

.emotion-11:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-11[aria-disabled=true] {
  color: #2e2e2e;
}

.emotion-11,
.emotion-11:hover,
.emotion-11:visited {
  background-color: #73b61c;
}

.emotion-11:hover {
  background-color: #96c955;
}

.emotion-11[aria-disabled=true] {
  background-color: #808080;
}

<form
    action=""
    className="emotion-13"
    method="post"
    onSubmit={[Function]}
  >
    <div
      className="emotion-3"
    >
      <label
        className="emotion-0"
        htmlFor="firstName"
      >
        First Name
      </label>
      <div
        className="emotion-2"
      >
        <input
          autoFocus={true}
          className="emotion-1"
          disabled={false}
          id="firstName"
          name="firstName"
          onChange={[Function]}
          type="text"
          value=""
        />
      </div>
      <div
        className="emotion-3"
      >
         
      </div>
    </div>
    <div
      className="emotion-3"
    >
      <label
        className="emotion-0"
        htmlFor="lastName"
      >
        Last Name
      </label>
      <div
        className="emotion-2"
      >
        <input
          className="emotion-1"
          disabled={false}
          id="lastName"
          name="lastName"
          onChange={[Function]}
          type="text"
          value=""
        />
      </div>
      <div
        className="emotion-3"
      >
         
      </div>
    </div>
    <div
      className="emotion-12"
    >
      <button
        className="emotion-10"
        onClick={[Function]}
        type="button"
      >
        Cancel
      </button>
      <button
        aria-disabled={true}
        className="emotion-11"
        onClick={[Function]}
        type="submit"
      >
        Save
      </button>
    </div>
  </form>,
]
`;

exports[`<AccountProfileForm /> should render properly when a user edits their name 3`] = `
Array [
  .emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding-top: 8px;
  padding-bottom: 8px;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  background-color: #d7f2de;
  border-radius: 2px;
  padding: 12px;
}

.emotion-0 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  padding-left: 12px;
  color: #191919;
  font-weight: 500;
}

.emotion-0 a {
  color: #4a90e2;
}

<div
    className="emotion-2"
  >
    <div
      className="emotion-1"
    >
      <svg
        color="#4ac565"
        data-file-name="checkmark.svg"
        height={20}
      />
      <div
        className="emotion-0"
        role="alert"
      >
        New name saved.
      </div>
    </div>
  </div>,
  .emotion-0 {
  font-size: 17px;
  line-height: 20px;
  font-weight: 500;
  padding-top: 16px;
}

<h3
    className="emotion-0"
  >
    Email: jane.doe@zorroa.com
  </h3>,
  .emotion-0 {
  padding-top: 24px;
  font-size: 17px;
  line-height: 20px;
  font-weight: 500;
}

<div
    className="emotion-0"
  >
    Name
  </div>,
  .emotion-0 {
  color: #808080;
  font-size: 15px;
  line-height: 19px;
  font-weight: 400;
  padding-bottom: 24px;
}

<div
    className="emotion-0"
  >
    Jane Doe
  </div>,
  .emotion-0 {
  -webkit-flex: 0;
  -ms-flex: 0;
  flex: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-radius: 2px;
  margin: 0;
  padding: 12px 32px;
  white-space: nowrap;
  font-size: 15px;
  line-height: 19px;
  font-weight: 500;
  border: none;
  cursor: pointer;
  color: #ffffff;
  background-color: transparent;
}

.emotion-0:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-0[aria-disabled=true] {
  color: #2e2e2e;
}

.emotion-0,
.emotion-0:hover,
.emotion-0:visited {
  background-color: #73b61c;
}

.emotion-0:hover {
  background-color: #96c955;
}

.emotion-0[aria-disabled=true] {
  background-color: #808080;
}

<button
    className="emotion-0"
    onClick={[Function]}
    type="button"
  >
    Edit
  </button>,
]
`;
